### PR TITLE
[SPARK-17038][Streaming] fix metrics retrieval source of 'lastReceivedBatch'

### PR DESCRIPTION
**Pull request by keypointt**
https://issues.apache.org/jira/browse/SPARK-17038
## What changes were proposed in this pull request?

StreamingSource's lastReceivedBatch_submissionTime, lastReceivedBatch_processingTimeStart, and lastReceivedBatch_processingTimeEnd all use data from lastCompletedBatch instead of lastReceivedBatch. 

In particular, this makes it impossible to match lastReceivedBatch_records with a batchID/submission time.

This is apparent when looking at StreamingSource.scala, lines 89-94.
## How was this patch tested?

Manually running unit tests on local laptop

### CODE REVIEW

File: streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala [hunk 1]
```diff
@@ -68,6 +68,7 @@ class StreamingJobProgressListenerSuite extends TestSuiteBase with Matchers {
     listener.waitingBatches should be (List(BatchUIData(batchInfoSubmitted)))
     listener.runningBatches should be (Nil)
     listener.retainedCompletedBatches should be (Nil)
+    listener.lastReceivedBatch should be (Some(BatchUIData(batchInfoSubmitted)))
```
Comment (jaceklaskowski) [inline streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala]: What do you think about `should contain (BatchUIData(batchInfoSubmitted))` instead? See http://www.scalatest.org/user_guide/using_matchers#workingWithContainers.

**Comment by srowen** [comment]
LGTM

File: streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala (hunk 1 above)
Comment (keypointt) [inline streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala]: I'm just following https://github.com/keypointt/spark/blob/ccd211f5d37d8d9c2d44511a766090aa6095a463/streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala#L129, to keep the style consistent with others

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.0.

**Pull request closed without merging**